apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -225,11 +225,128 @@ type PreferredClusterSelector struct {
+
+const (
+	// FromMinimum instructs Fleet to perform linear interpolation of weights starting
+	// from the minimum value, i.e., when the minimum value is most preferred.
+	//
+	// For example, for a preferred range [10, 30] with a minimum weight of 10 and a maximum
+	// weight of 100, Fleet will assign:
+	// * a weight of 100 to the cluster with the minimum value in the range; and
+	// * a weight of 10 to the cluster with the maximum value in the range; and
+	// * a weight of 37 to the cluster with a value of 16 in the range.
+	//
+	//   It is calculated using the formula below:
+	//   100-(16-10)/(30-10)*(100-10) = 73
+	FromMinimum InterpolationOrder = "FromMinimum"
+	// FromMaximum instructs Fleet to perform linear interpolation of weights starting
+	// from the maximum value, i.e., when the maximum value is most preferred.
+	//
+	// For example, for a preferred range [10, 30] with a minimum weight of 10 and a maximum
+	// weight of 100, Fleet will assign:
+	// * a weight of 10 to the cluster with the minimum value in the range; and
+	// * a weight of 100 to the cluster with the maximum value in the range; and
+	// * a weight of 37 to the cluster with a value of 16 in the range.
+	//
+	//   It is calculated using the formula below:
+	//   10+(16-10)/(30-10)*(100-10) = 37
+	FromMaximum InterpolationOrder = "FromMaximum"
+	// DoNotInterpolate instructs Fleet not to perform linear interpolation; when this
+	// option is used, alls clusters with a value in the preferred range will be assigned
+	// the same weight.
+	//
+	// This is the default option.
+	DoNotInterpolate InterpolationOrder = "DoNotInterpolate"
+)
+
+// MetricRange describes a required or preferred range for a metric.
+type MetricRange struct {
+	// Minimum is the minimum value of the range.
+	// If this value is not specified, the smallest observed value of the metric will
+	// be used.
+	// At least one of the minimum or maximum value must be specified.
+	// +optional
+	Minimum string `json:"minimum"`
+	// Maximum is the maximum value of the range.
+	// If this value is not specified, the largest observed value of the metric will be
+	// used.
+	// At least one of the minimum of maximum value must be specified.
+	// +optional
+	Maximum string `json:"maximum"`
+	// Interpolate is the interpolation order for the metric range.
+	//
+	// Specify this value if you would like Fleet to perform linear interpolation of weights
+	// for clusters with a metric value in the given range.
+	//
+	// Use the FromMinimum option if you would like to perfer clusters with a lower
+	// metric value; on the contrast, use the FromMaximum option if you would like to
+	// prefer clusters with a higher metric value.
+	//
+	// The default option is DoNotInterpolate, which means that all clusters with a
+	// metric value in the preferred range will be assigned the same weight.
+	// +optional
+	// +kubebuilder:default=DoNotInterpolate
+	Interpolate InterpolationOrder `json:"interpolate"`
```

Comment:
I think we still need to make sure this is signed off

Reply:
This is the option C form; should I ping anyone for approval?

Reply:
yes, please schedule a short meeting with Patrick. In the meantime, let's assume this is the base case.

Reply:
As an update, Patrick is on vacation and will be back on Mar 4; I have pinged Ahmed about this and will seek his inputs as well.

Reply:
No longer applicable after latest rounds of discussions; closing the thread.